packages/browser-data-collector/src/report.ts

```diff
@@ -1,48 +1,55 @@
 /**
  * Internal dependencies
  */
-import { getNavigationStart } from './api/performance-timing';
-import { deviceMemory, performanceTiming, environment } from './collectors';
+import {
+	deviceMemory,
+	performanceTiming,
+	environment,
+	fullPageStart,
+	inlineStart,
+	pageVisibilityStart,
+	pageVisibilityStop,
+} from './collectors';
 
 export class ReportImpl implements Report {
 	id: string;
-	start!: number;
+	beginning!: number;
 	end?: number;
-	data: ReportData;
-	collectors!: Collector[];
+	data: ReportData = new Map();
+	startCollectors!: Collector[];
+	stopCollectors!: Collector[];
 
-	constructor( id: string, isInitial: boolean ) {
+	static async fromNow( id: string ) {
+		const report = new ReportImpl( id, false );
+		await report.start();
+		return report;
+	}
+
+	static async fromPageStart( id: string ) {
+		const report = new ReportImpl( id, true );
+		await report.start();
+		return report;
+	}
```

Comment:
Workaround because JavaScript doesn't support `async new Report(...)`

Reply:
Async constructors seem like a very odd concept to me; a constructor should always return an object of that type, not a promise nor anything else. These async static methods are great, though 👍